src/common/pf_lldp.c

```diff
@@ -440,10 +527,167 @@ void pf_lldp_send(
 void pf_lldp_init(
    pnet_t                  *net)
 {
+   memset(&net->fspm_cfg.lldp_peer_cfg, 0, sizeof(net->fspm_cfg.lldp_peer_cfg));
+
    pf_lldp_send(net);
    if (pf_scheduler_add(net, PF_LLDP_INTERVAL*1000,
       lldp_sync_name, pf_lldp_trigger_sending, NULL, &net->lldp_timeout) != 0)
    {
       LOG_ERROR(PF_ETH_LOG, "LLDP(%d): Failed to reschedule LLDP sending\n", __LINE__);
    }
 }
+
+
+void pf_lldp_recv(
+    pnet_t              *net,
+    os_buf_t            *p_lldp_buffer,
+    uint16_t            offset)
+{
+   char                 alias[PNET_LLDP_PORT_ID_MAX_LEN+PNET_LLDP_PORT_ID_MAX_LEN+2];
+   lldp_frame_t         frame = {0};
+   pnet_lldp_peer_cfg_t *lldp_peer_cfg = &net->fspm_cfg.lldp_peer_cfg;
+   pf_get_info_t        parse_info;
+
+   LOG_DEBUG(PF_LLDP_LOG, "LLDP(%d): packet received len=%d, offset=%d\n", __LINE__, p_lldp_buffer->len, offset);
+
+   parse_info.result = PF_PARSE_OK;
+   parse_info.is_big_endian = true;
+   parse_info.len = p_lldp_buffer->len;
+   parse_info.p_buf = (uint8_t *)p_lldp_buffer->payload;
+
+   LLDP_DUMP_BUF(&parse_info.p_buf[offset], parse_info.len - offset);
+
+   frame = pf_get_lldp_tlv_header(&parse_info, &offset);
+
+   while (frame.type != LLDP_TYPE_END)
+   {
+      LOG_DEBUG(PF_LLDP_LOG, "LLDP(%d): lldp tlv type=%d len=%zu\n", __LINE__, frame.type, frame.len);
+
+      switch (frame.type)
+      {
+      case LLDP_TYPE_CHASSIS_ID:
+         if ((frame.len > 0) &&
+             (frame.len - 1) < sizeof(lldp_peer_cfg->chassis_id))
+         {
+            lldp_peer_cfg->chassis_id_len = frame.len - 1;
+            lldp_peer_cfg->chassis_id_subtype = pf_get_byte(&parse_info, &offset);
+            pf_get_mem(&parse_info, &offset, lldp_peer_cfg->chassis_id_len, &lldp_peer_cfg->chassis_id);
+            lldp_peer_cfg->chassis_id[lldp_peer_cfg->chassis_id_len] = '\0';
+         }
+         else
+         {
+            LOG_ERROR(PF_LLDP_LOG, "LLDP(%d): Chassis Id too long (%zu)\n", __LINE__, frame.len - 1);
+         }
+         break;
+      case LLDP_TYPE_PORT_ID:
+         if ((frame.len > 0) &&
+             (frame.len - 1) < sizeof(lldp_peer_cfg->port_id))
+         {
+            lldp_peer_cfg->port_id_len = frame.len - 1;
+            lldp_peer_cfg->port_id_subtype = pf_get_byte(&parse_info, &offset);
+            pf_get_mem(&parse_info, &offset, lldp_peer_cfg->port_id_len, &lldp_peer_cfg->port_id);
+            lldp_peer_cfg->chassis_id[lldp_peer_cfg->port_id_len] = '\0';
+
+            /* Update Alias name as follows:
```

Comment:
Implement a function to generate the alias name.
```
/**
 * @internal
 * Generate an alias name from port_id and chassis_id
 *
 * Defined in Profinet 2.4, section 4.3.1.4.18
 *
 * Check if the LLDP_TYPE_PORT_ID contains a "." (Example: port-001.test)
 *  If it does copy this over to the alias name
 *  If no "." is found concatenate the port id and chassis id
 *
 * @param port_id          In:   Peer port ID. Terminated string.
 * @param chassis_id       In:   Peer chassis ID. Terminated string.
 * @param alias            Out:  Alias name. Terminated string.
 * @param len              In:   Size of output buffer
 * @return  0  if the operation succeeded.
 *          -1 if an error occurred.
 */
int pf_lldp_generate_alias_name(
const char*             port_id,
const char*             chassis_id,
char*             	alias,
uint16_t                len)
{
}
```


Implement unit tests. Use: 
`class LldpUnitTest : public PnetUnitTest {};`

See for example:
`pf_cmina_is_stationname_valid()`

Reply:
Ok,
I split the receive operation in a parser and an action function.